### PR TITLE
Add StyleListener to ClientTheme to fix lightModeFixes

### PR DESCRIPTION
**Pull request by CodeF53**
Light mode fixes of client theme replaces all `--white-500` backgrounds with `--primary-100`, but since discord started lazy loading css chunks, the old approach of generating all these fixes at `DOMContentLoaded` is no longer functional.

This PR adds a mutationObserver to the head of the DOM to generate new fixes for lightMode as the app is used.

### CODE REVIEW

File: src/plugins/clientTheme/cssUtil.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+async function parseStyleLinkNode(node: HTMLLinkElement): Promise<string> {
+    const link = node.getAttribute("href");
+    if (!link) return "";
+
+    try {
+        const res = await fetch(link);
+        return await res.text();
+    } catch (error) {
+        return "";
+    }
```
Comment (Vendicated) [inline src/plugins/clientTheme/cssUtil.ts:42]:
```suggestion
    return fetch(link)
    	.then(res => res.text())
        .catch(() => "");
```

File: src/plugins/clientTheme/cssUtil.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ */
+
+// returns all of discord's native styles in a single string
+// ! discord lazy loads css chunks, make sure to use newStyleListener as well
+export async function getStyles(): Promise<string> {
+    let out = "";
+    const styleLinkNodes: NodeListOf<HTMLLinkElement> = document.querySelectorAll('link[rel="stylesheet"]');
+    for (const styleLinkNode of styleLinkNodes)
+        out += await parseStyleLinkNode(styleLinkNode);
+    return out;
```
Comment (Vendicated) [inline src/plugins/clientTheme/cssUtil.ts:14]:
```suggestion
    return Promise.all(styleLinkNodes.map(n => parseStyleLinkNode(n)))
    	.then(contents => contents.join("\n"));
```

File: src/plugins/clientTheme/cssUtil.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+// runs callback with styles of new style nodes added to the head
+export function newStyleListener(callback: (styles: string) => void) {
+    const observer = new MutationObserver(mutations => {
```
Comment (Vendicated) [inline src/plugins/clientTheme/cssUtil.ts:19]: i'm unsure if using a MutationObserver is a good idea. css chunks are loaded via webpack (`wreq.f.css`).

it'd be better to hook into that instead

this func is used for all except the initial chunk
image
image
https://discord.com/assets/12633.bb59c3b8d0e17096489a.css

File: src/plugins/clientTheme/cssUtil.ts
Comment (CodeF53) [inline src/plugins/clientTheme/cssUtil.ts:1]: @Vendicated using tabs in suggested change, evil

File: src/plugins/clientTheme/cssUtil.ts (hunk 3 above)
Comment (CodeF53) [reply src/plugins/clientTheme/cssUtil.ts:18]: I will ask for advice on how to patch `wreq.f.css` when I have time to fix pr

**Comment by Vendicated** [comment]
lint explosion

File: src/plugins/clientTheme/cssUtil.ts (hunk 3 above)
Comment (CodeF53) [reply src/plugins/clientTheme/cssUtil.ts:18]: you can't regex patch the webpack loader, how do I hook into wreq.f.css then?

**Pull request closed without merging**